core/state/statedb.go

```diff
@@ -1314,7 +1336,7 @@ func (s *StateDB) Commit(block uint64, deleteEmptyObjects bool) (common.Hash, er
 // - Add coinbase to access list (EIP-3651)
 // - Reset transient storage (EIP-1153)
 func (s *StateDB) Prepare(rules params.Rules, sender, coinbase common.Address, dst *common.Address, precompiles []common.Address, list types.AccessList) {
-	if rules.IsBerlin {
+	if rules.IsEIP2929 {
```

Comment:
Why this change?

Reply:
I can answer that! 
Because the clause is needed `when eip 2929 is active`. That is, after `Berlin`, so `IsBerlin`, but not if verkle is enabled. So we _could_ do, `if rules.Berlin && !rules.Verkle`. Or, we could make verkle-transition un-set `IsEip2929`, which I prefer. So essentially, verkle would undo 2929.